addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1331,83 +1329,11 @@ options.registry.OptionsTab = options.Class.extend({
+        await this.customizeWebsiteVariable(previewMode, widgetValue ? `'${widgetValue}'` : '', { variable: 'body-image' });
     },
-    /**
-     * @see this.selectClass for parameters
-     */
     async openCustomCodeDialog(previewMode, widgetValue, params) {
-        const libsProm = loadBundle({
-            jsLibs: [
-                '/web/static/lib/ace/ace.js',
-                '/web/static/lib/ace/mode-xml.js',
-                '/web/static/lib/ace/mode-qweb.js',
-            ],
-        });
-
-        let websiteId;
-        this.trigger_up('context_get', {
-            callback: (ctx) => {
-                websiteId = ctx['website_id'];
-            },
-        });
-
-        let website;
-        const dataProm = this._rpc({
-            model: 'website',
-            method: 'read',
-            args: [[websiteId], ['custom_code_head', 'custom_code_footer']],
-        }).then(websites => {
-            website = websites[0];
-        });
-
-        let fieldName, title, contentText;
-        if (widgetValue === 'head') {
-            fieldName = 'custom_code_head';
-            title = _t('Custom head code');
-            contentText = _t('Enter code that will be added into the <head> of every page of your site.');
-        } else {
-            fieldName = 'custom_code_footer';
-            title = _t('Custom end of body code');
-            contentText = _t('Enter code that will be added before the </body> of every page of your site.');
-        }
-
-        await Promise.all([libsProm, dataProm]);
-
-        await new Promise(resolve => {
-            const $content = $(core.qweb.render('website.custom_code_dialog_content', {
-                contentText,
-            }));
-            const aceEditor = this._renderAceEditor($content.find('.o_ace_editor_container')[0], website[fieldName] || '');
-            const dialog = new Dialog(this, {
-                title,
-                $content,
-                buttons: [
-                    {
-                        text: _t("Save"),
-                        classes: 'btn-primary',
-                        click: async () => {
-                            await this._rpc({
-                                model: 'website',
-                                method: 'write',
-                                args: [
-                                    [websiteId],
-                                    {[fieldName]: aceEditor.getValue()},
-                                ],
-                            });
-                        },
-                        close: true,
-                    },
-                    {
-                        text: _t("Discard"),
-                        close: true,
-                    },
-                ],
-            });
-            dialog.on('closed', this, resolve);
-            dialog.open();
-        });
+        this.trigger_up('open_edit_head_body_dialog');
```

Comment:
Method options should only resolve their promise when the related action is finished. Here you trigger_up... then it is immediately synchronously finished. This is not ok because it means that other option can have their effect triggered as this one is considered finished. In general, this also means that your CTRL-Z entry for this option won't work (in this case as it is a setting in DB, it probably does not matter but...).

The common way to handle this with a trigger_up would be to give a function to call when the called action is done and have that function resolve a local promise defined here. Something like:
```javascript
return new Promise(resolve => {
    this.trigger_up('open_edit_head_body_dialog', {
        onSuccess: resolve,
    });
});
```